lib/panmaster/selftest/src/testcases/panrequest_cb_test.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either expess or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <uwb/uwb.h>
+#include "panmaster_test.h"
+#include "pan.h"
+
+TEST_CASE_SELF(panrequest_cb_test)
+{
+    int rc;
+    int i;
+
+    struct pan_req_resp *req = calloc(sizeof(*req), 1);
+    struct pan_req_resp *rsp = calloc(sizeof(*rsp), 1);
+    struct test_pan_mock_instance *mock_pan = calloc(sizeof(*mock_pan), 1);
+    static struct panmaster_node_idx node_idx[MYNEWT_VAL(PANMASTER_MAXNUM_NODES)];
+
+    for (i = 0; i < MYNEWT_VAL(PANMASTER_MAXNUM_NODES); i++) {
+        PANMASTER_NODE_IDX_DEFAULT(node_idx[i]);
+    }
+
+    node_idx[0].addr = 0x29A;
+    node_idx[0].role = 1;
+
+    node_idx[1].addr = 0x29B;
+    node_idx[1].role = 1;
```

Comment:
You should probably have the callback add these to the node_idx instead of doing it manually. When the find function fail to find it a new entry should be created to hold the new client.

Reply:
Maybe now  the way of adding nodes is correct?